### ISSUE TITLE
Failed to start executable - "caRepeater"

### ISSUE THREAD

**Issue by HongYi-Shern** [pyepics/pyepics#183]
Failed to start executable - "caRepeater"
Changes may be required in your "path" environment variable.
caStartRepeaterIfNotInstalled (): unable to start CA repeater daemon detached process

What do i need to add to the environment variable? No problem at first test

Thank you

**Comment by newville** [comment]
@HongYi-Shern This is a benign message from the CA library.   

Basically, if caReapeter can be found in your path it will be launched as a background service by the routines in libca.  If you build epics base for your system, that will build such a caReapeter executable.   There is no real problems caused by having it not run, but some potential efficiencies in having it run on machines that will have multiple processes communicating to the same CA servers.

**Comment by ndevenish** [comment]
Is there some way to suppress this message?

**Comment by newville** [comment]
@ndevenish  Just to be clear, these messages come from the CA library, not from Python.  As earlier, this message is "mostly harmless"  [D. Adams], pointing out that "caRepeater" was not found in your PATH.  Having "caRepeater" in your PATH is generally recommended but not required.

You can suppress or redirect *all* of the messages from libCA, including the "Virtual circuit disconnect" warnings when an IOC shuts down, with something like:

```
import epics
def ignore_ca_messages(*args):  pass

epics.ca.replace_printf_handler(ignore_ca_messages) 
```

That is sort of a blunt tool, suppressing all messages.

**Comment by ndevenish** [comment]
Thanks, in my case I'm running a cluster service and specifically using pyepics so that I _don't_ have to worry about installing/having the command line tools available. I'd rather avoid unnecessary logged output, especially that looks like an error!

**Comment by newville** [comment]
@ndevenish Yeah,  "suppressing all warnings" is not something one normally recommends.  ;). 
But the messages from libCA are the kind that you want off most of the time, but still be able to turn on for debugging.

Perhaps pyepics should swallow these messages by default but allow them to be enabled?
